@orlp Interestingly, those were already covered.

```python
s1 = pl.Series("a", [1, 200000, 300000], dtype=pl.Int32)
s2 = pl.Series("b", [1, 2, 3], dtype=pl.Int8)

print(s1 | s2)
```

gives:

```
Traceback (most recent call last):
  File "/home/amber/git/scratch/test.py", line 34, in <module>
    print(s1 | s2)
          ~~~^~~~
  File "/home/amber/git/polars-2/py-polars/src/polars/series/series.py", line 813, in __or__
    return self._from_pyseries(self._s.bitor(other._s))
                               ~~~~~~~~~~~~~^^^^^^^^^^
polars.exceptions.SchemaError: cannot unpack series of type `i8` into `i32'
```

--- a/crates/polars-plan/src/plans/aexpr/schema.rs
+++ b/crates/polars-plan/src/plans/aexpr/schema.rs
@@ -640,15 +640,14 @@ fn get_arithmetic_field(
             if (left_field.dtype.is_bool() ^ right_field.dtype.is_bool())
                 && (left_field.dtype.is_numeric() ^ right_field.dtype.is_numeric()) =>
         {
-            let supertype = try_get_supertype(&left_field.dtype, &right_field.dtype)?;
-            polars_warn!(
-                Deprecation,
-                "{op} on {:?} and {:?} is deprecated and will raise a ComputeError in Polars 2.0\n\
-                Hint: cast the Boolean to {supertype:?} using pl.Expr.cast().",
+            polars_bail!(
+                ComputeError:
+                "{op} on {:?} and {:?} is not supported\n\
+                Hint: cast the Boolean to {:?} using pl.Expr.cast().",
                 left_field.dtype,
                 right_field.dtype,
+                try_get_supertype(&left_field.dtype, &right_field.dtype)?,
             );
-            supertype
         },
         _ => {
             match (&left_field.dtype, &right_field.dtype) {

--- a/py-polars/tests/unit/operations/arithmetic/test_arithmetic.py
+++ b/py-polars/tests/unit/operations/arithmetic/test_arithmetic.py
@@ -1069,7 +1069,7 @@ def test_log_broadcast(dtype: pl.DataType) -> None:
     [(operator.and_, "&"), (operator.or_, "|"), (operator.xor, "^")],
     ids=["and", "or", "xor"],
 )
-def test_bitwise_bool_ops_deprecated(
+def test_bitwise_bool_ops_unsupported(
     op: Callable[[Any, Any], Any], op_str: str
 ) -> None:
     lf = pl.LazyFrame(
@@ -1077,18 +1077,12 @@ def test_bitwise_bool_ops_deprecated(
     )
     hint = "Hint: cast the Boolean to Int32 using pl.Expr.cast()."
 
-    msg = (
-        f"{op_str} on Boolean and Int32 is deprecated and will raise a ComputeError in Polars 2.0\n"
-        + hint
-    )
-    with pytest.warns(DeprecationWarning, match=rf"^{re.escape(msg)}$"):
+    msg = f"{op_str} on Boolean and Int32 is not supported\n" + hint
+    with pytest.raises(pl.exceptions.ComputeError, match=rf"^{re.escape(msg)}"):
         lf.select(op(pl.col("bool"), pl.col("int"))).collect_schema()
 
-    msg = (
-        f"{op_str} on Int32 and Boolean is deprecated and will raise a ComputeError in Polars 2.0\n"
-        + hint
-    )
-    with pytest.warns(DeprecationWarning, match=rf"^{re.escape(msg)}$"):
+    msg = f"{op_str} on Int32 and Boolean is not supported\n" + hint
+    with pytest.raises(pl.exceptions.ComputeError, match=rf"^{re.escape(msg)}"):
         lf.select(op(pl.col("int"), pl.col("bool"))).collect_schema()
 
 

--- a/py-polars/tests/unit/operations/map/test_inefficient_map_warning.py
+++ b/py-polars/tests/unit/operations/map/test_inefficient_map_warning.py
@@ -32,8 +32,8 @@ TEST_CASES = [
     # ---------------------------------------------
     ("a", "lambda x: x + 1 - (2 / 3)", '(pl.col("a") + 1) - 0.6666666666666666', None),
     ("a", "lambda x: x // 1 % 2", '(pl.col("a") // 1) % 2', None),
-    ("a", "lambda x: x & True", 'pl.col("a") & True', None),
-    ("a", "lambda x: x | False", 'pl.col("a") | False', None),
+    ("g", "lambda x: x & True", 'pl.col("g") & True', None),
+    ("g", "lambda x: x | False", 'pl.col("g") | False', None),
     ("a", "lambda x: abs(x) != 3", 'pl.col("a").abs() != 3', None),
     ("a", "lambda x: int(x) > 1", 'pl.col("a").cast(pl.Int64) > 1', None),
     (
@@ -371,6 +371,7 @@ def test_parse_apply_functions(
                 datetime(2024, 5, 6),
                 datetime(2077, 10, 20),
             ],
+            "g": [True, False, None],
         }
     )
 

--- a/py-polars/tests/unit/operations/test_selectors.py
+++ b/py-polars/tests/unit/operations/test_selectors.py
@@ -1156,12 +1156,12 @@ def test_bitwise_ops_eval_on_expr(df: pl.DataFrame) -> None:
     with pytest.raises(
         pl.exceptions.InvalidOperationError, match="& not allowed on time and u16"
     ):
-        df.select(cs.all() & pl.col("abc"))
+        df.select(cs.time() & pl.col("abc"))
     with pytest.raises(
         pl.exceptions.InvalidOperationError, match=r"\| not allowed on time and u16"
     ):
-        df.select(cs.all() | pl.col("abc"))
+        df.select(cs.time() | pl.col("abc"))
     with pytest.raises(
         pl.exceptions.InvalidOperationError, match=r"\^ not allowed on time and u16"
     ):
-        df.select(cs.all() ^ pl.col("abc"))
+        df.select(cs.time() ^ pl.col("abc"))

